protoc-gen-twirp/generator.go

```diff
@@ -819,7 +819,7 @@ func (t *twirp) generateUtils() {
 
 	t.P(`// doJSONRequest makes a JSON request to the remote Twirp service.`)
 	t.P(`func doJSONRequest(ctx `, t.pkgs["context"], `.Context, client HTTPClient, hooks *`, t.pkgs["twirp"], `.ClientHooks, url string, in, out `, t.pkgs["proto"], `.Message) (_ `, t.pkgs["context"], `.Context, err error) {`)
-	t.P(`  marshaler := &`, t.pkgs["protojson"], `.MarshalOptions{UseProtoNames: true}`)
+	t.P(`  marshaler := &`, t.pkgs["protojson"], `.MarshalOptions{UseProtoNames: !s.jsonCamelCase, EmitUnpopulated: !s.jsonSkipDefaults}`)
```

Comment:
I don't think this would compile. The `s` receiver (server) is not accessible on this function. This function is used by the client to send requests, and the client constructor currently [doesn't have options](https://github.com/twitchtv/twirp/blob/main/client_options.go) like the ones for the server [WithServerJSONSkipDefaults](https://github.com/twitchtv/twirp/pull/372/files) and [WithServerJSONCamelCaseNames](https://github.com/twitchtv/twirp/blob/main/server_options.go#L73).

To verify this, check the instructions on [CONTRIBUTING.md](https://github.com/twitchtv/twirp/blob/main/CONTRIBUTING.md#contributing-code-with-pull-requests) on how to re-generate the code and run tests with the newly generated code.

Reply:
Thanks for the info. I will check on how to fix this issue properly and improve my pull request according to the instructions.